### PR TITLE
Add release notes for snapshots

### LINKED ISSUE #243: Publish Snapshot builds of master
cc @shivaram

### PR DESCRIPTION
**Pull request by shivaram**
Fixes #243

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+## KeystoneML release process
+
+1. We publish KeystoneML releases, snapshots to sonatype repository https://oss.sonatype.org/. To
```
Comment (etrain) [inline RELEASE.md]: s/releases, snapshots/releases and snapshots/

File: RELEASE.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+3. Add `sbt-pgp` and `sbt-sonatype` to your plugins at `~/.sbt/0.13/plugins/gpg.sbt`
+```
+addSbtPlugin("com.jsuereth" % "sbt-pgp" % "1.0.0")
+
+addSbtPlugin("org.xerial.sbt" % "sbt-sonatype" % "1.1")
+```
+
+4. Add your Sonatype credentials to `~/.sbt/0.13/sonatype.sbt` 
+credentials += Credentials("Sonatype Nexus Repository Manager",
```
Comment (etrain) [inline RELEASE.md]: can we put this in a code block? ```scala works

**Comment by etrain** [comment]
Two minor comments, otherwise LGTM.

**Comment by shivaram** [comment]
Done

**Pull request merged**